lib/components/mp4muxer/index.ts

```diff
@@ -22,89 +23,140 @@ export class Mp4Muxer extends Tube {
     const onSync = (ntpPresentationTime: number) => {
       this.onSync && this.onSync(ntpPresentationTime)
     }
+    let sdpHasIncompleteH264Props = false
+    let clonedSdpMessage = messageFromBuffer(Buffer.from(''))
+    let haveSdpMessage = false
+    let cachedSps = Buffer.from('')
+    let haveSps = false
+    let cachedPps = Buffer.from('')
+    let havePps = false
+    let sentMoov = false
     const incoming = new Transform({
       objectMode: true,
       transform: function (msg: Message, encoding, callback) {
         if (msg.type === MessageType.SDP) {
           /**
            * Arrival of SDP signals the beginning of a new movie.
-           * Set up the ftyp and moov boxes.
+           * Set up the ftyp and moov boxes unless sdpHasIncompleteH264Props is true.
            */
-
-          // Why is this here? These should be default inside the mvhd box?
-          const now = Math.floor(new Date().getTime() / 1000 + 2082852000)
-          const ftyp = new Box('ftyp')
-          const moov = boxBuilder.moov(msg.sdp, now)
-
-          const data = Buffer.allocUnsafe(ftyp.byteLength + moov.byteLength)
-          ftyp.copy(data, 0)
-          moov.copy(data, ftyp.byteLength)
-
-          debug('msl:mp4:isom')(`ftyp: ${ftyp.format()}`)
-          debug('msl:mp4:isom')(`moov: ${moov.format()}`)
-
-          // Set up a list of tracks that contain info about
-          // the type of media, encoding, and codec are present.
-          const tracks = msg.sdp.media.map((media) => {
-            return {
-              type: media.type,
-              encoding: media.rtpmap && media.rtpmap.encodingName,
-              mime: media.mime,
-              codec: media.codec,
+          msg.sdp.media.forEach((media) => {
+            if (media.rtpmap === undefined) {
+              return
+            }
+            if (media.rtpmap.encodingName === 'H264') {
+              sdpHasIncompleteH264Props = true
+              if (media.fmtp === undefined) {
+                return
+              }
+              if (media.fmtp.parameters !== undefined &&
+                media.fmtp.parameters['profile-level-id'] !== undefined &&
+                media.fmtp.parameters['sprop-parameter-sets'] !== undefined) {
+                //console.log("sprops:" + media.fmtp.parameters['sprop-parameter-sets'])
```

Comment:
another statement that could be added to the debug?

Reply:
For later perhaps...